commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/service/impl/CommercePriceEntryLocalServiceImpl.java

```diff
@@ -615,6 +615,28 @@ public CommercePriceEntry upsertCommercePriceEntry(
 			}
 		}
 
+		CPInstance cpInstance = null;
+
+		if (cProductId > 0) {
+			cpInstance = _cpInstanceLocalService.fetchCPInstance(cProductId);
```

Comment:
why cProductId ?

Reply:
Hi @marco-leo , I was following the upsert pattern, so first try to find the entity by Id, in this case cProductId is the cpInstanceId. This is something that may happen with the API upsert request, so i thought it was right to manage this case instead of let the DuplicatException arise. Please let me know if it is something it should be changed.

Thank you

Reply:
@andrea-ale-sbarra I think your logic does make sense -- what doesn't make sense is that `cProductId` is actually the name of the primary key of the `CProduct` table, so using it here as the primary key of `CPInstance` is confusing. If the variable always represents the primary key of `CPInstance`, we should rename it to `cpInstanceId`; if this variable is really a `cProductId` then this service call will always return `null` (because we don't reusing primary key values); if this variable could represent either one then we should rename it to something more generic -- we use `productId` or even just `id` for things like this elsewhere.

Reply:
Hi @ethanbustad as far as I can see this method is used only in the PriceList headless module and this variable gets its value with this piece of code `GetterUtil.getLong(priceEntry.getSkuId())` and `skuId` is defined in the DTOConverter like this `skuId = cpInstance.getCPInstanceId();`, so I share your opinion, this variable could have been luckier in the naming, I think it should be renamed.

Thank you @ethanbustad for your inputs!